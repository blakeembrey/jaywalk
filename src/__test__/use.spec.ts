import { validationFunction } from '../'

describe('#use', () => {
  it('should validate using use property', () => {
    const validate = validationFunction({
      '@type': 'String',
      use: [
        {
          '@type': 'String',
          minLength: 2
        },
        {
          '@type': 'String',
          minLength: 4
        }
      ]
    })

    const input = 'testing'

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject invalid use schemas', () => {
    const validate = validationFunction({
      '@type': 'String',
      use: [
        {
          '@type': 'String',
          minLength: 10
        }
      ]
    })

    const input = 'testing'

    expect.assertions(1)

    return validate(input).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
