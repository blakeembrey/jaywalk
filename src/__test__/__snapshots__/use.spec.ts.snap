// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#use should reject invalid use schemas 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "min length",
    "schemaPath": Array [
      "use",
      0,
      "minLength",
    ],
  },
]
`;
